src/main/java/util/Parser.java

```diff
@@ -0,0 +1,174 @@
+package util;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+/**
+ * Processor and executor for user inputs.
+ */
+public class Parser {
+
+    /**
+     * Runs the given input string if valid.
+     * Prints any output or error messages.
+     *
+     * @param input String input read from user.
+     * @param taskList List of task stored by driver.
+     * @param ui UI printer containing standard print methods.
+     * @throws DukeException If input is invalid.
+     */
+    public static String parse(String input, TaskList taskList, Ui ui) throws DukeException {
+
+        assert taskList != null && ui != null;
+
+        String output = "";
+
+        if (input.equals("list")) {
+
+            output += taskList.printList();
+
+        } else if (input.indexOf("done ") == 0) {
+
+            String[] arr = input.split(" ");
+            int index = Integer.parseInt(arr[1]) - 1;
+            Task task = taskList.completeTask(index);
+            output += "Nice! I've marked this task as done:\n";
+            output += task + "\n";
+
+        } else if (input.indexOf("delete ") == 0) {
+
+            String[] arr = input.split(" ");
+            int index = Integer.parseInt(arr[1]) - 1;
+            Task task = taskList.removeTask(index);
+            output += "Noted. I've removed this task:\n";
+            output += task + "\n";
+            output += "Now you have " + taskList.getSize() + " tasks in the list.\n";
+
+        } else if (input.indexOf("find ") == 0) {
+            output = find(input, output, taskList, ui);
+        } else {
+            output = addToList(input, output, taskList, ui);
+        }
+
+        output += ui.getLine();
+
+        return output;
+    }
+
+    private static String find(String input, String output, TaskList taskList, Ui ui) throws DukeException {
+
+        if (input.contains("/on ")) {
+
+            int index = input.indexOf("/on ");
+
+            String dateString = input.substring(index + 4);
```

Comment:
Can the magic number be converted to a more meaningful value? (e.g. `"/on ".length()`)

![image](https://user-images.githubusercontent.com/53135010/92476146-1641d380-f211-11ea-936d-bd98788cd62d.png)